node/pkg/processor/vaa_processor.go

```diff
@@ -0,0 +1,352 @@
+	signedInC chan *gossipv1.SignedVAAWithQuorum
+	// sendC is a channel of outbound messages to broadcast on p2p
+	sendC chan []byte
+
+	attestationEvents *reporter.AttestationEventReporter
+	logger            *zap.Logger
+	db                *db.Database
+
+	manager *reactor.Manager[*vaa.VAA]
+
+	// gst is managed by the processor and allows concurrent access to the
+	// guardian set by other components.
+	gst *common.GuardianSetState
+
+	governor        *governor.ChainGovernor
+	pythnetVaas     map[string]PythNetVaaEntry
+	pythnetVAAsLock sync.Mutex
+}
+
+func NewVAAReactor(
+	db *db.Database,
+	lockC chan *common.MessagePublication,
+	setC chan *common.GuardianSet,
+	sendC chan []byte,
+	obsvC chan *gossipv1.SignedObservation,
+	obsvReqSendC chan<- *gossipv1.ObservationRequest,
+	signedInC chan *gossipv1.SignedVAAWithQuorum,
+	gk *ecdsa.PrivateKey,
+	gst *common.GuardianSetState,
+	attestationEvents *reporter.AttestationEventReporter,
+	g *governor.ChainGovernor,
+) *VAAReactor {
+	obsC := make(chan *vaa.VAA, 10)
+
+	r := &VAAReactor{
+		obsvReqSendC:      obsvReqSendC,
+		signedInC:         signedInC,
+		lockC:             lockC,
+		obsC:              obsC,
+		sendC:             sendC,
+		gst:               gst,
+		db:                db,
+		attestationEvents: attestationEvents,
+		governor:          g,
+	}
+
+	manager := reactor.NewManager[*vaa.VAA]("vaa", obsC, obsvC, setC, gst, reactor.Config{
+		RetransmitFrequency: 5 * time.Minute,
+		QuorumGracePeriod:   2 * time.Minute,
+		QuorumTimeout:       24 * time.Hour,
+		UnobservedTimeout:   24 * time.Hour,
+		Signer:              reactor.NewEcdsaKeySigner(gk),
+		NetworkAdapter:      reactor.NewChannelNetworkAdapter(sendC),
+	}, r, r)
+	r.manager = manager
+
+	return r
+}
+
+func (p *VAAReactor) Run(ctx context.Context) error {
+	p.logger = supervisor.Logger(ctx)
+
+	err := supervisor.Run(ctx, "vaa-reactor-manager", p.manager.Run)
+	if err != nil {
+		return fmt.Errorf("failed to start reactor manager: %w", err)
+	}
+
+	cleanup := time.NewTicker(30 * time.Second)
+	defer cleanup.Stop()
+
+	// Always initialize the timer so don't have a nil pointer in the case below. It won't get rearmed after that.
+	govTimer := time.NewTicker(time.Minute)
+	defer govTimer.Stop()
+
+	for {
+		select {
+		case <-ctx.Done():
+			return ctx.Err()
+		case k := <-p.lockC:
+			// Ignore incoming observations when our database already has a quorum VAA for it.
+			// This can occur when we're receiving late observations due to node catchup.
+			if existing, err := p.getSignedVAA(*db.VaaIDFromVAA(k.CreateVAA(0))); err == nil {
+				p.logger.Debug("ignoring observation since we already have a quorum VAA for it",
+					zap.Stringer("emitter_chain", k.EmitterChain),
+					zap.Stringer("emitter_address", k.EmitterAddress),
+					zap.Uint32("nonce", k.Nonce),
+					zap.Stringer("txhash", k.TxHash),
+					zap.Time("timestamp", k.Timestamp),
+					zap.String("message_id", existing.MessageID()),
+				)
+				continue
```

Comment:
It does get checked and queued up in the governor in L127.
For existing signed VAAs that should not matter, should it? Otherwise we'd need also pass VAAs received from others to the governor in `handleSignedVAA`